.gitignore

```diff
@@ -1 +1,12 @@
+# IDE created folders
+# -------------------
+.idea
+
+# CMAKE folders
+# -------------
+cmake-build-debug
```

Comment:
cmake-build-debug also CLion IDE specific (cmake-build-* would be better)
I'd prefere not see comments here or make it realistic.